packages/config/src/reader.ts

```diff
@@ -83,9 +83,27 @@ export class ConfigReader implements Config {
     // Merge together all configs into a single config with recursive fallback
     // readers, giving the first config object in the array the lowest priority.
     return configs.reduce<ConfigReader>(
-      (previousReader, { data, context, filteredKeys }) => {
+      (previousReader, { data, context, filteredKeys, deprecatedKeys }) => {
         const reader = new ConfigReader(data, context, previousReader);
         reader.filteredKeys = filteredKeys;
+
+        // TODO(cmpadden) introduce `deprecatedKeys` and `notifiedDeprecatedKeys` & use logger
+        if (deprecatedKeys) {
+          for (const { key, description } of deprecatedKeys) {
+            if (description) {
```

Comment:
I think we could merge these x) Perhaps just append if description is present? `` `removed soon. ${description || ''}` ``

Would be good to also include the context here btw, helps identify where the deprecated value is defined

Reply:
✅ Combined and added context!